### PR TITLE
Merging 2.7.0 to 2.8.0

### PR DESCRIPTION
**Pull request by harishgilimi**

### CODE REVIEW

File: src/app/client/projects/desktop/src/app/modules/content-consumption/components/content-player/content-player.component.ts [hunk 1]
```diff
@@ -76,9 +76,10 @@ export class ContentPlayerComponent implements AfterViewInit, OnChanges, OnInit,
     this.contentRatingModal = false;
     if (!_.isEmpty(this.playerConfig)) {
       this.objectRollUp = _.get(this.playerConfig, 'context.objectRollup') || {};
-      this.loadPlayer();
+        this.loadPlayer();
+        this.handleYoutubeContent(this.contentData);
```
Comment (vinukumar-vs) [inline src/app/client/projects/desktop/src/app/modules/content-consumption/components/content-player/content-player.component.ts]: Why this change? Is it verified in both Portal & Desktop?

File: src/app/client/projects/desktop/src/app/modules/content-consumption/components/content-player/content-player.component.ts [hunk 2]
```diff
@@ -159,10 +160,15 @@ export class ContentPlayerComponent implements AfterViewInit, OnChanges, OnInit,
    * Emits event when content starts playing and end event when content was played/read completely
    */
   loadPlayer() {
-    if (_.includes(this.router.url, 'browse')) {
+    const downloadStatus = _.has(this.playerConfig, 'metadata.desktopAppMetadata') ?
+    !_.has(this.playerConfig, 'metadata.desktopAppMetadata.isAvailable') ||
```
Comment (vinukumar-vs) [inline src/app/client/projects/desktop/src/app/modules/content-consumption/components/content-player/content-player.component.ts]: Define all these string in the CONSTANT variable & use it. This is to avoid hardcoded strings in the code.

File: src/app/client/projects/desktop/src/app/modules/content-consumption/components/content-player/content-player.component.ts [hunk 3]
```diff
@@ -159,10 +160,15 @@ export class ContentPlayerComponent implements AfterViewInit, OnChanges, OnInit,
...
+    _.get(this.contentData, 'desktopAppMetadata.isAvailable') : false;
+
+    if (_.includes(this.router.url, 'browse') && !downloadStatus) {
```
Comment (vinukumar-vs) [inline src/app/client/projects/desktop/src/app/modules/content-consumption/components/content-player/content-player.component.ts]: why do you need these changes here? Pass it as part of PlayerConfig & check.

**Comment by harishgilimi** [comment]
@vinukumar-vs  Above components are part of desktop project not belongs to portal code

**Comment by vinukumar-vs** [comment]
Some of the changes of Programs portal code doesn't have required quality gate code coverage. But it has addressed in R2.8.
Hence merging this code.
